File: cognite/pygen/_client/auth/credentials.py
"""Base credentials interface for authentication."""

from abc import ABC, abstractmethod


class Credentials(ABC):
    """
    Abstract base class for authentication credentials.

    All credential types must inherit from this class and implement
    the get_headers method to provide authentication headers for HTTP requests.
    """

    @abstractmethod
    def authorization_header(self) -> tuple[str, str]:
        """
        Get the authorization header for HTTP requests.

        Returns:
            A tuple containing the header name and header value.
        """
        pass

File: cognite/pygen/_typescript/instance_api/auth/credentials.ts
/**
 * Base credentials interface for authentication.
 *
 * All credential types must implement this interface to provide
 * authentication headers for HTTP requests.
 *
 * @packageDocumentation
 */

/**
 * Abstract interface for authentication credentials.
 *
 * Credential implementations must provide a method to get
 * the authorization header for HTTP requests.
 */
export interface Credentials {
  /**
   * Get the authorization header for HTTP requests.
   *
   * @returns A promise resolving to a tuple of [header name, header value]
   *
   * @example
   * ```_typescript
   * const [name, value] = await credentials.authorizationHeader();
   * headers.set(name, value);
   * ```
   */
  authorizationHeader(): Promise<[string, string]>;

  /**
   * Close any resources held by the credentials.
   * This is optional - credentials that don't hold resources don't need to implement it.
   */
  close?(): void;
}
